### PR TITLE
support array with scalar arithmetic operation for decimal data type

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

part of #122

```
schema:
CREATE EXTERNAL TABLE food (a DECIMAL(10,5), b decimal(20,15), c boolean) xxxx;
```
This query `select a+b from table` was supported, but the query `select a+10 from table` is not supported.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/physical-expr/src/expressions/binary.rs
Comment (alamb) [inline datafusion/physical-expr/src/expressions/binary.rs:283]: It would be great to move some / all of this decimal manipulation into the arrow-rs crate at some point

**Review by alamb — APPROVED** [review]
Looks good @liukun4515

File: datafusion/physical-expr/src/expressions/binary.rs
Comment (liukun4515) [reply datafusion/physical-expr/src/expressions/binary.rs:283]: thanks for your reminder and I have a plan to do this.

Comment (liukun4515) [reply datafusion/physical-expr/src/expressions/binary.rs:283]: But now I find a bug in the arithmetic operation. https://github.com/apache/arrow-datafusion/blob/c91efc27658e58264c4f346a5cfdec8810179e90/datafusion/physical-expr/src/expressions/binary.rs#L302 and https://github.com/apache/arrow-datafusion/blob/c91efc27658e58264c4f346a5cfdec8810179e90/datafusion/physical-expr/src/expressions/binary.rs#L307

In the divide method, we may lose of precision by converting the `i128` to `f64`.
We use the `f64` to store the larger range of data, but we lose the precision.

I will try to figure out this issue and resolve this.

**Comment by liukun4515** [comment]
I think It can be merged

**Comment by alamb** [comment]
> I think It can be merged


Sounds good to me -- feel free to do it yourself next time; I normally wait for at least one approving review from a maintainer or active contributor. If the change is trivial I'll merge it then and if a little more involved I try and leave it open for 24 hours.

**Comment by alamb** [comment]
Thanks again @liukun4515

**Pull request merged**
